# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1298595951 18000
#      Thu Feb 24 20:05:51 2011 -0500
# Node ID 009068efc66dd3d93835fa136542b4cba49d5e16
# Parent  26d3164fd58d8c1f69eebb7d48c2dbfe33593595
Properly evaluate zeroth order polynomials.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2011-02-24  Robert T. Short  <rtshort@ieee.org> 
+
+	* polynomial/polyval.m: Properly evaluate constant polynomials.
+
 2011-02-24  John W. Eaton  <jwe@octave.org>
 
 	* strings/strchr.m: Avoid implicit string to number conversion.
 	Bug #32546.  From Kim Hansen <kim@i9.dk>.
 
 2011-02-22  Ben Abbott  <bpabbott@mac.com>
 
 	* plot/subplot.m: Use new looseinset property introduced by changeset
diff --git a/scripts/polynomial/polyval.m b/scripts/polynomial/polyval.m
--- a/scripts/polynomial/polyval.m
+++ b/scripts/polynomial/polyval.m
@@ -64,17 +64,17 @@ function [y, dy] = polyval (p, x, s, mu)
 
   if (length (p) == 0)
     y = p;
     return;
   endif
 
   n = length (p) - 1;
   x = (x - mu(1)) / mu(2);
-  y = p(1);
+  y = p(1) * ones (size (x(:)));
   for i = 2:n+1
     y = y .* x(:) + p(i);
   endfor
   y = reshape (y, size (x));
 
   if (nargout == 2)
     ## Note: the F-Distribution is generally considered to be single-sided.
     ## http://www.itl.nist.gov/div898/handbook/eda/section3/eda3673.htm
@@ -123,8 +123,24 @@ endfunction
 %! assert (x, polyval(p,x), eps)
 %! x = reshape(x, [2, 5]);
 %! assert (x, polyval(p,x), eps)
 %! x = reshape(x, [5, 2]);
 %! assert (x, polyval(p,x), eps)
 %! x = reshape(x, [1, 1, 5, 2]);
 %! assert (x, polyval(p,x), eps)
 
+%!test
+%! p = [1];
+%! x = 1:10;
+%! y = ones(size(x));
+%! assert (y, polyval(p,x), eps)
+%! x = x(:);
+%! y = ones(size(x));
+%! assert (y, polyval(p,x), eps)
+%! x = reshape(x, [2, 5]);
+%! y = ones(size(x));
+%! assert (y, polyval(p,x), eps)
+%! x = reshape(x, [5, 2]);
+%! y = ones(size(x));
+%! assert (y, polyval(p,x), eps)
+%! x = reshape(x, [1, 1, 5, 2]);
+
